cocos/platform/android/java/src/org/cocos2dx/lib/Cocos2dxActivity.java

```diff
@@ -355,7 +371,7 @@ public void onWindowFocusChanged(boolean hasFocus) {
     }
 
     private void resumeIfHasFocus() {
-        if(hasFocus) {
+        if(hasFocus && !paused) {
```

Comment:
Why it is needed? `resumeIfHasFocus()` is invoked by `onResome()`, and `paused` is set to false.

Reply:
in the scenario that I described, below steps happen.
1. onWindowFocusChanged(false) is called when swiping down notification drawer.
2. onPause() is called when locking the screen. -> This pauses all audios.
3. onWindowFocusChanged(true) is called right after 2. -> call resumeIfHasFocus() -> This resumes all audios even though the screen has been locked.

So it seems to be needed to check pause status.

I verified this scenario on Google Pixel XL and Google Pixel 2.

detail Cocos2dxActivity logs
D/Cocos2dxActivity: onWindowFocusChanged() hasFocus=false
D/Cocos2dxActivity: onPause()
D/Cocos2dxActivity: onWindowFocusChanged() hasFocus=true
D/Cocos2dxActivity: onWindowFocusChanged() hasFocus=false